src/Graphics/Image/Processing/Filter.hs

```diff
@@ -140,3 +141,31 @@ prewittOperator !img = sqrt (prewittX ^ (2 :: Int) + prewittY ^ (2 :: Int))
         !prewittY = applyFilter (prewittFilter Vertical Edge) img
 {-# INLINE prewittOperator #-}
 
+
+-- |The Laplacian of an image highlights regions of rapid intensity change 
+-- and is therefore often used for edge detection. It is often applied to an 
+-- image that has first been smoothed with something approximating a 
+-- Gaussian smoothing filter in order to reduce its sensitivity to noise.
+--
+-- <<images/yield.jpg>>   <<images/yield_laplacian.png>> 
+-- 
+-- Usage : 
+--	
+-- >>> img <- readImageY VU "images/yield.jpg"
+-- >>> let laplacianImage :: Image VU RGB Double
+-- >>>     laplacianImage = applyFilter (laplacianFilter Complete Edge) img
+-- >>> writeImage "images/yield_laplacian.png" (toImageRGB laplacianImage)
+--
+laplacianFilter :: (Array arr cs e, Array arr X e) =>
+                   Direction -> Border (Pixel cs e) -> Filter arr cs e
+laplacianFilter dir !border =
+  Filter (correlate border kernel)
+  where
+    !kernel =
+      case dir of
+        Complete   -> fromLists $ [ [ -1, -1, -1 ]     -- Unlike the Sobel edge detector, the Laplacian edge detector uses only one kernel. 
```

Comment:
It's a bit annoying that nothing but `Complete` can be handled by `laplacianFilter`, and conversely the filters who use the other `Direction` constructors will not be able to handle `Complete`. So there's always a risk of runtime error if you add `Complete` there. Also, what is the point of doing it if that's the only option that `laplacianFilter` can handle anyway?

Reply:
Ok sir, will do so.